### ISSUE TITLE
Error during install: "websocket Appears to be a git repo or submodule"

### ISSUE THREAD

**Issue by spalladino** [OpenZeppelin/openzeppelin-sdk#929]
Creating a new issue based on @crazyrabbitLTC's https://github.com/zeppelinos/zos/issues/926.

> When running Node v10.16.0 this error was appearing on installs of ZOS and Truffle.
> 
>     Dennisons-MacBook-Pro:my-project dennison$ npm install zos
>     npm ERR! path /Users/dennison/Documents/my-project/node_modules/websocket
>     npm ERR! code EISGIT
>     npm ERR! git /Users/dennison/Documents/my-project/node_modules/websocket: Appears to be a git repo or submodule.
>     npm ERR! git     /Users/dennison/Documents/my-project/node_modules/websocket
>     npm ERR! git Refusing to remove it. Update manually,
>     npm ERR! git or move it out of the way first.
>     
>     npm ERR! A complete log of this run can be found in:
>     npm ERR!     /Users/dennison/.npm/_logs/2019-05-31T18_56_24_258Z-debug.log
> 
> This is an error that @nventuro told me happens to him also frequently and that the only solution appears to be running rm -rf node_modules and then doing npm install again.

I have not been able to reproduce this on a fresh install in node 10.16. Based on the message, it would seem like you already had something in your `node_modules` folder. Can you identify the steps you need to run on an empty folder to get to that error?

~By the way, it would seem like running just `rm -rf node_modules/*/.git/` (see [this comment](https://github.com/react-native-community/react-native-safe-area-view/issues/73#issuecomment-492971839)) fixes the issue. It should be a faster workaround than clearing all of `node_modules`.~

**Comment by frangio** [comment]
I can reproduce this error by simply running the following command in an empty directory.

```
npm install github:frozeman/WebSocket-Node.git#browserifyCompatible
```

This is the same dependency that is causing the error. Running the command will create the `node_modules/websocket/.git` directory, and trying to run the command once more will fail with the same error.

This is npm's fault as far as I can tell. However, the package/repo would be able to mitigate this if it specified ` "files"` in package.json, or `.npmignore`d the `.git` directory.

I am using node 11.15.0 and npm 6.9.0.

**Comment by frangio** [comment]
Official comment by the web3.js maintainer on this situation: https://github.com/ethereum/web3.js/issues/2863#issuecomment-498032927.

**Comment by frangio** [comment]
I would not recommend `rm -rf node_modules/*/.git/`. If there's a local dependency installed there will be a symlink and its `.git` directory would be removed.

**Comment by spalladino** [comment]
Thanks @frangio, we will wait on web3 for the fix then

**Comment by itinance** [comment]
Unfortunately the issue on web3 (https://github.com/ethereum/web3.js/issues/2863#issuecomment-498032927) was already closed, without providing any solution. It breaks circle-ci as well with node 9 container

**Comment by spalladino** [comment]
Fixed as web3 was upgraded
